rosbag2_examples/rosbag2_examples_cpp/src/simple_bag_recorder.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021 Open Source Robotics Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+
+#include "example_interfaces/msg/string.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+#include "rosbag2_cpp/writer.hpp"
+
+using std::placeholders::_1;
+
+class SimpleBagRecorder : public rclcpp::Node
+{
+public:
+  SimpleBagRecorder()
+  : Node("simple_bag_recorder")
+  {
+    writer_ = std::make_unique<rosbag2_cpp::Writer>();
+
+    writer_->open("my_bag");
+
```

Comment:
It seems missed `writer_->create_topic(..)`

Reply:
`rosbag2_cpp::Writer` automatically creates topics when you write to them (it calls `SequentialWriter::create_topic` internally if necessary). This is a simpler API and I prefer to leave it the way it is written